### ISSUE TITLE
Distributed locking with redis sentinel does not honor server address configuration property

### ISSUE THREAD

**Issue by flavioschuindt** [Netflix/conductor#2642]
**Describe the bug**
 I am trying to enable a distributed locking using Redis sentinel. Due to the lack of missing information in documentation, I am looking directly in the code and I identified these properties that needs to be configured. They are coming from interfaces  ```Configuration``` and ```RedisLock```:
 
 ```properties
#enable locking during workflow execution
 workflow.decider.locking.enabled=true
 workflow.decider.locking.namespace=conductor
 workflow.decider.locking.server=REDIS
 workflow.decider.locking.leaseTimeInSeconds=60
 ```

```properties
#Redis sentinel settings for locking module
workflow.redis.locking.server.type=SENTINEL
workflow.redis.locking.server.master.name=mymaster
```

```properties
#Comma seperated list of server nodes
workflow.redis.locking.server.address=redis://<MY_HOST>:26379 # Of course, I replace <MY_HOST> to the proper host string.
workflow.redis.locking.server.password=<MY_PASSWORD> # Of course, I replace <MY_PASSWORD> to the proper password string.
```

Not sure if relevant here, but my redis sentinel is deployed on a kubernetes cluster using the redis sentinel chart from bitnami. I see the deployment is up and running. I can access normally redis and all SENTINEL commands works fine.

With the above configuration, I see that the Redis Module is loaded, but I keep receiving this error when I start a workflow execution:

```Redis url should start with redis:// or rediss:// (for SSL connection)``` (This message is coming from Redisson ```RedisURI```class).

Investiganting further, I see that above error coming from Redisson. One interesting thing is that Redisson (```SentinelConnectionManager``` class) outputs this:

```terminal
266888 [pool-15-thread-5] INFO  org.redisson.connection.SentinelConnectionManager  - master: null://conductor-redis-node-0.conductor-redis-headless.wo.svc.cluster.local:6379 added
266980 [pool-15-thread-5] INFO  org.redisson.connection.SentinelConnectionManager  - slave: null://conductor-redis-node-2.conductor-redis-headless.wo.svc.cluster.local:6379 added
266980 [pool-15-thread-5] INFO  org.redisson.connection.SentinelConnectionManager  - slave: null://conductor-redis-node-1.conductor-redis-headless.wo.svc.cluster.local:6379 added
```
Note how the scheme is defined as null. I don't understand why because I am clearly setting the ```workflow.redis.locking.server.address``` property in conductor containing ```redis://``` as schema. It seems that conductor is not properly forwarding this to Redisson.

**Details**
Conductor version: 2.30.3
Persistence implementation: Postgress
Queue implementation: Postgres
Lock: Redis
Workflow definition: N/A
Task definition: N/A
Event handler definition: N/A

**To Reproduce**
1. Enable Redis locking with above mentioned properties
2. See the error coming in conductor logs

**Expected behavior**
I would expect that the ```redis://``` schema is honored in the configuration and no error message is printed in the logs.

**Screenshots**
N/A

**Additional context**
Both conductor and redis are running in a Kubernetes cluster.

**Comment by flavioschuindt** [comment]
Ok, looks like I found the issue. The issue is in redisson. It's very easy to reproduce:

* Create a simple java app;
* Add redisson in the classpath;
* Copy and paste the same block of code that conductor is using to connect to redis sentinel:

```java
Config redisConfig = new Config();
redisConfig.useSentinelServers()
		.setScanInterval(2000)
		.setMasterName("mymaster")
		.addSentinelAddress("redis://localhost:7002") // This is a redis sentinel up and running on my machine
		.setPassword("<PASSWORD>") // This is the redis sentinel password
		.setTimeout(10000);
RedissonClient redisson = Redisson.create(redisConfig);
```
* In this simple Java app, make sure you are referring in the pom.xml the exact samve redisson version that conductor is currently using (3.11.4);
* Run the app. You will see the exact same error mentioned above: ```Redis url should start with redis:// or rediss:// (for SSL connection)```.

This issue seems to be fixed in redisson ```3.12.1```:

* Bump the version in pom.xml to 3.12.1;
* Error goes away and redisson successfully connects in the redis sentinel.

Now, the problem is that I was checking and looks like the redisson version of conductor was bumped in this [commit](https://github.com/Netflix/conductor/blob/2de2f4b6e3d855a9306a08024199472800ea17f0/redis-lock/dependencies.lock), but none of 2.X version has it. Should this be backported to 2.X too? I can raise a PR. Thoughts, @kishorebanala, @aravindanr?

P.S.: Unfortunately, going to a 3.X version is not an option for me at this moment.

**Comment by aravindanr** [comment]
@flavioschuindt Thanks for posting the details. Please create a PR to bump up the version in 2.31 branch.

**Comment by flavioschuindt** [comment]
Done in #2651, @aravindanr. Could you please review and merge? Also, once merged, do you have an idea of the timeline to be added in 2.31.6? Our team is dependent on this and we would like, of course, to consume directly from the open source project. Thank you!

**Comment by apanicker-nflx** [comment]
Closed by #2651
